src/software/ai/hl/stp/play/dynamic_plays/offense_play.cpp

```diff
@@ -0,0 +1,134 @@
+    }
+
+    // Determine number of defense and support tactics to assign
+    auto [num_defenders, num_supporters] = assignNumOfDefendersAndSupporters(
+        std::max(static_cast<int>(play_update.num_tactics) - 1, 0));
+
+    // Get defense tactics from DefensePlay
+    std::vector<std::shared_ptr<Tactic>> defense_tactics;
+    defense_play_->updateTactics(PlayUpdate(
+        play_update.world_ptr, num_defenders,
+        [&](PriorityTacticVector new_tactics)
+        {
+            for (auto& tactic_vec : new_tactics)
+            {
+                defense_tactics.insert(defense_tactics.end(), tactic_vec.begin(),
+                                       tactic_vec.end());
+            }
+        },
+        play_update.inter_play_communication,
+        play_update.set_inter_play_communication_fun));
+
+    // Create more support tactics if necessary
+    // (so that `support_tactics_` has `num_supporters` in it)
+    updateSupportTactics(num_supporters);
+
+    if (!support_tactics_.empty())
+    {
+        SkillState skill_state = attacker_tactic_->getSkillState();
+
+        std::vector<Point> existing_receiving_positions;
+        auto support_tactics_it = support_tactics_.begin();
+
+        // If the attacker has committed to a pass, set the receiving position
+        // of the first support tactic to the receiver point of the pass
+        if (skill_state.pass_committed && skill_state.pass)
+        {
+            Point receiving_position = skill_state.pass->receiverPoint();
+            support_tactics_.front()->updateReceivingPosition(receiving_position);
+            existing_receiving_positions.push_back(receiving_position);
+            ++support_tactics_it;
+        }
+
+        unsigned int num_positions_to_generate = static_cast<unsigned int>(
+            support_tactics_.size() - existing_receiving_positions.size());
+
+        // Generate receiving positions for the remaining support tactics
+        std::vector<Point> receiving_positions = strategy->getBestReceivingPositions(
+            num_positions_to_generate, existing_receiving_positions);
+
+        for (const Point& receiving_position : receiving_positions)
+        {
+            (*support_tactics_it)->updateReceivingPosition(receiving_position);
+            ++support_tactics_it;
+        }
+    }
+
+    // Add all defense and support tactics to the main PriorityTacticVector
+    TacticVector secondary_tactics;
+    secondary_tactics.insert(secondary_tactics.end(), support_tactics_.begin(),
+                             support_tactics_.end());
+    secondary_tactics.insert(secondary_tactics.end(), defense_tactics.begin(),
+                             defense_tactics.end());
+    tactics.push_back(secondary_tactics);
+
+    play_update.set_tactics(tactics);
+}
+
+std::tuple<unsigned int, unsigned int> OffensePlay::assignNumOfDefendersAndSupporters(
+    unsigned int num_tactics)
+{
+    unsigned int num_defenders, num_supporters;
+    switch (num_tactics)
```

Comment:
shouldn't we encounter for the attacker that's assigned

Reply:
No need to account for the attacker in this method since we pass in the number of defenders/supporters we need to assign. We get that number by subtracting 1 from the total # of tactics (if the attacker is present) in `updateTactics`